conncheck_go19.go

```diff
@@ -0,0 +1,53 @@
+// Go MySQL Driver - A MySQL-Driver for Go's database/sql package
+//
+// Copyright 2012 The Go-MySQL-Driver Authors. All rights reserved.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// +build go1.9
+
+package mysql
+
+import (
+	"errors"
+	"io"
+	"net"
+	"syscall"
+)
+
+var errUnexpectedRead = errors.New("unexpected read from socket")
+
+func connCheck(c net.Conn) error {
+	var (
+		n int
+		err error
+		buff [1]byte
+	)
+
+	sconn, ok := c.(syscall.Conn)
+	if !ok {
+		return nil
+	}
+	rc, err := sconn.SyscallConn()
+	if err != nil {
+		return err
+	}
+	rerr := rc.Read(func(fd uintptr) bool {
+		n, err = syscall.Read(int(fd), buff[:])
+		return true
```

Comment:
Have you tested it on Windows?

Reply:
I have not! As pointed out in the PR body:

> To note: I have not been able to test this patch in Windows, so I'm not sure whether it works there or whether the conncheck should be disabled for that platform.

Do you have any ideas on how would be able to run the CI test on Windows?

Reply:
It's difficult to add it in short term.
Instead, you can test your code in small repository, like this.
https://github.com/methane/peek/runs/25901610

In my case, I stopped while thinking about nice API works on both of Windows and Unix.